packages/@aws-cdk/aws-cloudwatch/lib/dashboard.ts

```diff
@@ -67,6 +68,10 @@ export class Dashboard extends Resource {
       physicalName: props.dashboardName,
     });
 
+    if (props.dashboardName && !props.dashboardName.match(/^[\w-]+$/)) {
```

Comment:
I thought about replacing the `+` quantifier by a `{1,255}` to match [the length requirement](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloudwatch-dashboard.html#cfn-cloudwatch-dashboard-dashboardname), but that would probably also require a different error message